plugins/modules/zos_copy.py

```diff
@@ -609,14 +686,38 @@ def copy_to_seq(
             dest {str} -- Name of destination data set
             src_ds_type {str} -- The type of source
             alloc_vol {str} -- The volume where destination should be allocated
+            type {str} -- Type of data set, if it needs created
+            space_primary {int} -- Primary allocation, if data set needs created
+            space_secondary {int} -- Secondary allocation, if data set needs created
+            space_type {str} - Units of measure (CYL/TRK/K/M) of allocation values
+            record_format {str} - Format (FB/VB) of data set if data set needs created
+            record_length {int} - Size of individual records in the record set
+            block_size {int} - Size of data block
         """
         new_src = temp_path or conv_path or src
-        if model_ds:
-            if self.dest_exists:
-                datasets.delete(dest)
-            self.allocate_model(dest, model_ds, vol=alloc_vol)
+        # Pre-clear data set
+        if self.dest_exists:
+            datasets.delete(dest)
+            self.dest_exists = False
```

Comment:
Need to discuss, because there are a few options here, but we don't have a way for the user to indicate it:
1 - make destination looks like source:
        If source is dataset, it is straightforward to copy source attributes
        If source is USS, we need parameters or some other modeling (what model_ds was for)
2 - destination looks like pre-existing destination
        We could update the parameters to look like destination before deletion, but should this override any parameters sent in?
        Those parameters have default values, which could make this trickier, unless params are ONLY for non-pre-existing destination.
3 - destination looks like parameters
        Straightforward, but conflicts with 2 and 1b
        This would almost make parameters a requirement, which I don't think we want

User can't indicate if destination should or should not change format.  Jerry's question last week needed the destination to stay the way he had pre-allocated it, which would be an argument for #2.

Reply:
Lets have a call tomorrow, I also thought Jerry's case was more about not deleting a destination if they passed it in , might need to be reminded so Ill go read the JIRA tomorrow.

Reply:
- DS parms always take precedence - if users go to the trouble to set up DS parms let them rule over all cases.
- If dest exists we use the dest as the model DS
- if dest does not exist:
   a) if the src is a DS , and gave us NO parms, we use src as the model
- if the src is USS and Dest does not exist (and no DS parms were provided)
 a) we will try to give them a DS with defaults for their dest and hope it works else fail

Reply:
Thanks for working on these changes and collaborating !!!